src/MongoDB.Driver/Linq/Linq3Implementation/Translators/ExpressionToAggregationExpressionTranslators/ConstantExpressionToAggregationExpressionTranslator.cs

```diff
@@ -26,7 +28,26 @@ public static AggregationExpression Translate(TranslationContext context, Consta
             if (expression.NodeType == ExpressionType.Constant)
             {
                 var value = expression.Value;
-                var valueSerializer = BsonSerializer.LookupSerializer(expression.Type); // TODO: use known serializer
+                var defaultValueSerializer = expression.Type switch
+                {
+                    Type t when t == typeof(bool) => new BooleanSerializer(),
+                    Type t when t == typeof(string) => new StringSerializer(),
+                    Type t when t == typeof(byte) => new ByteSerializer(),
+                    Type t when t == typeof(short) => new Int16Serializer(),
+                    Type t when t == typeof(ushort) => new UInt16Serializer(),
+                    Type t when t == typeof(int) => new Int32Serializer(),
+                    Type t when t == typeof(uint) => new UInt32Serializer(),
+                    Type t when t == typeof(long) => new Int64Serializer(),
+                    Type t when t == typeof(ulong) => new UInt64Serializer(),
+                    Type t when t == typeof(float) => new SingleSerializer(),
+                    Type t when t == typeof(double) => new DoubleSerializer(),
+                    Type t when t == typeof(decimal) => new DecimalSerializer(),
+                    Type { IsConstructedGenericType: true } t when t.GetGenericTypeDefinition() == typeof(Nullable<>) => (IBsonSerializer)Activator.CreateInstance(typeof(NullableSerializer<>).MakeGenericType(t.GenericTypeArguments[0])),
+                    Type { IsArray: true } t => (IBsonSerializer)Activator.CreateInstance(typeof(ArraySerializer<>).MakeGenericType(t.GetElementType())),
+                    _ => null
+                };
+                // If we need an EnumSerializer, we have to look up the correct one based on the current expression.
+                var valueSerializer = context.KnownSerializersRegistry.GetSerializer(expression, defaultValueSerializer);
```

Comment:
I'm quite certain that we need a default. Consider the following examples:

```
var result = Project(x => new { Result = x.G.ElementAt(1).D });
var result = Project(x => new { Result = x.J.ToString("%Y-%m-%d") });
var result = Project(x => new { Result = x.C.E.I.Except(new[] { "it", "not in here" }) });
```

In each case we need to find a serializer for a method argument. There is no serializer in the tree that tells us how to serialize these arguments. If you can think of something clever to figure it out, but backstopping constants by default serializers is the solution that I came up with. I agree that it is possible that it could hide legitimate bugs, but I couldn't think of something better.

Reply:
I'm still not convinced we should have a default.

I understand how the current implementation is using the default, but I think having a default is dangerous.

But we can come back to this (maybe) some time in the future.